src/main/scala/beam/sim/BeamMobsim.scala

```diff
@@ -91,6 +94,82 @@ class BeamMobsim @Inject()(
     logger.info("Processing Agentsim Events (End)")
   }
 
+  private def clearRoutesIfNeeded(iteration: Int): Unit = {
+    val experimentType = beamServices.beamConfig.beam.physsim.relaxation.`type`
+    if (experimentType == "experiment_2.0" && beamServices.beamConfig.beam.physsim.relaxation.experiment2_0.clearRoutesEveryIteration) {
+      clearRoutes()
+      logger.info(s"Experiment_2.0: Clear all routes at iteration ${iteration}")
+    } else if (experimentType == "experiment_2.1" && beamServices.beamConfig.beam.physsim.relaxation.experiment2_1.clearRoutesEveryIteration) {
+      clearRoutes()
+      logger.info(s"Experiment_2.1: Clear all routes at iteration ${iteration}")
+    } else if (experimentType == "experiment_3.0" && iteration <= 1) {
+      clearRoutes()
+      logger.info(s"Experiment_3.0: Clear all routes at iteration ${iteration}")
+    } else if (experimentType == "experiment_4.0" && iteration <= 1) {
+      clearRoutes()
+      logger.info(s"Experiment_4.0: Clear all routes at iteration ${iteration}")
+    } else if (experimentType == "experiment_5.0" && iteration <= 1) {
+      clearRoutes()
+      logger.info(s"Experiment_5.0: Clear all routes at iteration ${iteration}")
+    } else if (experimentType == "experiment_5.1" && iteration <= 1) {
+      clearRoutes()
+      logger.info(s"Experiment_5.1: Clear all routes at iteration ${iteration}")
+    } else if (experimentType == "experiment_5.2" && iteration <= 1) {
+      clearRoutes()
+      logger.info(s"Experiment_5.2: Clear all routes at iteration ${iteration}")
+    }
+  }
+
+  private def clearModesIfNeeded(iteration: Int): Unit = {
+    val experimentType = beamServices.beamConfig.beam.physsim.relaxation.`type`
+    if (experimentType == "experiment_2.0" && beamServices.beamConfig.beam.physsim.relaxation.experiment2_0.clearModesEveryIteration) {
+      clearModes()
+      logger.info(s"Experiment_2.0: Clear all modes at iteration ${iteration}")
+    } else if (experimentType == "experiment_2.1" && beamServices.beamConfig.beam.physsim.relaxation.experiment2_1.clearModesEveryIteration) {
+      clearModes()
+      logger.info(s"Experiment_2.1: Clear all modes at iteration ${iteration}")
+    } else if (experimentType == "experiment_3.0" && iteration <= 1) {
+      clearModes()
+      logger.info(s"Experiment_3.0: Clear all modes at iteration ${iteration}")
```

Comment:
would merging clearRoutesIfNeeded and clearModesIfNeeded lead to less code duplication?

Reply:
I merged them together to remove code duplication